src/applications/mhv-secure-messaging/tests/components/DeleteDraft.unit.spec.jsx

```diff
@@ -1,40 +1,144 @@
 import React from 'react';
-import { renderInReduxProvider } from '@department-of-veterans-affairs/platform-testing/react-testing-library-helpers';
+import { renderInReduxProvider } from 'platform/testing/unit/react-testing-library-helpers';
+import { Router } from 'react-router-dom';
+import { combineReducers, applyMiddleware, createStore } from 'redux';
+import thunk from 'redux-thunk';
+import { commonReducer } from 'platform/startup/store';
+import { createTestHistory } from 'platform/testing/unit/helpers';
+
 import { expect } from 'chai';
-import { fireEvent } from '@testing-library/react';
+import { fireEvent, waitFor } from '@testing-library/react';
 import sinon from 'sinon';
 import DeleteDraft from '../../components/Draft/DeleteDraft';
 import { drafts, inbox, sent } from '../fixtures/folder-inbox-response.json';
+import thread from '../fixtures/reducers/thread-with-multiple-drafts-reducer.json';
 import reducer from '../../reducers';
 import { Prompts } from '../../util/constants';
-import * as mockProps from '../fixtures/delete-draft-mock-prop.json';
 
 describe('Delete Draft component', () => {
-  it('renders without errors', async () => {
+  // This renderWithStoreAndRouter function is a replica function from the /platform-testing/react-testing-library-helpers
+  // file, the only difference is that 'store' is returned and used to mock the dispatch function in the tests.
+  // **Special permissions may be needed to alter the /platform-testing/react-testing-library-helpers file.
+  function renderWithStoreAndRouter(

```

Comment:
I can see that we are using this function for one dispatch validation, but it may not be specific enough (which action is actually dispatched? -  that is not actually being tested). There may be another approach used to test iff an action was dispatched. Refer to `ManageFolderButtons.unit.spec.jsx` in `confirming removal of a folder with no threads contained triggers a call` test. That way we may not need to introduce this duplicative code. However, I like an idea to return a store in that function, it can make some validations easier. We may consider contributing that in a separate PR